src/core/event_registry/server_event.rs

```diff
@@ -582,36 +663,36 @@ unsafe fn send_with<E: Event>(
 /// # Safety
 ///
 /// The caller must ensure that `event_data` was created for `E`.
-unsafe fn serialize_with<E: Event>(
+unsafe fn serialize_with_tick<E: Event>(
     event_data: &ServerEvent,
     ctx: &mut ServerSendCtx,
     event: &E,
-    client: &ReplicatedClient,
+    init_tick: RepliconTick,
```

Comment:
I refactored this function to pass tick instead of `ReplicatedClient` for clarity.